p2p/src/sync/tests/block_list_request.rs

```diff
@@ -62,9 +62,11 @@ async fn max_block_count_in_request_exceeded(#[case] seed: Seed) {
 
         let peer = node.connect_peer(PeerId::new(), protocol_version).await;
 
-        let blocks = iter::repeat(block.get_id())
-            .take(*p2p_config.protocol_config.max_request_blocks_count + 1)
-            .collect();
+        let blocks = iter::repeat_n(
+            block.get_id(),
+            *p2p_config.protocol_config.max_request_blocks_count + 1,
+        )
+        .collect();
         peer.send_block_sync_message(BlockSyncMessage::BlockListRequest(BlockListRequest::new(
```

Comment:
FYI: ironically, the original diagnostic from clippy was "this `repeat().take()` can be written more **concisely**". 
But now the same code is spread over 5 lines instead of 3, thanks to rustfmt.